src/domain/eventSearch/utils.tsx

```diff
@@ -263,30 +311,44 @@ export const getEventSearchVariables = ({
     keywordAnd.push('yso:p4354');
   }
 
-  const keywordsParamName = MAPPED_KEYWORD_TERMS[searchSource];
-  const mappedCategories: string[] = categories
-    .map((category) => MAPPED_CATEGORIES[searchSource][category])
-    .filter((e) => e);
-  const mappedHobbyTypes: string[] =
-    hobbyTypes
-      ?.map((hobbyType) => MAPPED_COURSE_HOBBY_TYPES[hobbyType])
-      .filter((e) => e) || [];
+  const categoriesParamName = MAPPED_KEYWORD_TERMS[searchSource];
+
+  const categoryMap = MAPPED_CATEGORIES[searchSource];
+
+  const getMappedPropertyValues = (
+    list: string[],
+    map: Record<string, string>
+  ) =>
+    list?.reduce<string[]>(
+      (prev, val: string) => prev.concat(map[val]?.split(',') ?? []),
+      []
+    );
+
+  const mappedCategories = getMappedPropertyValues(categories, categoryMap);
+
+  const mappedHobbyTypes = getMappedPropertyValues(
+    hobbyTypes ?? [],
+    MAPPED_COURSE_HOBBY_TYPES
+  );
+
```

Comment:
In this part there seems to be lot of spacing between code lines.. is it intentional? Looks little bit strange

Reply:
fixed